From c99028f88a63c19e71e0d46e6e846e40ba1e7820 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Tue, 29 Jun 2021 14:10:33 +0200
Subject: [PATCH] imx8mm-var-dart-nrt: Add led start routine for NRT

The gpio and i2c commands are provided in New OS for C 1 6-3-21 pdf

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 include/configs/imx8mm_var_dart.h | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/include/configs/imx8mm_var_dart.h b/include/configs/imx8mm_var_dart.h
index c8f6ac299e..15da1ec940 100644
--- a/include/configs/imx8mm_var_dart.h
+++ b/include/configs/imx8mm_var_dart.h
@@ -81,6 +81,14 @@
 	"m4_addr=0x7e0000\0" \
 	"m4_bin=hello_world.bin\0" \
 	"use_m4=no\0" \
+	"led_ctrl=0x32\0" \
+	"nrt_led_addrs1=0x1C 0x16 0x17 0x1D 0x18 0x19;\0" \
+	"nrt_led_addrs2=0x26 0x27 0x2C 0x28 0x29 0x2D;\0" \
+	"nrt_led_init=gpio set 11; gpio set 70; i2c dev 2;\0" \
+	"nrt_led_start=i2c mw ${led_ctrl} 0x0 0x40; i2c mw ${led_ctrl} 0x36 0x5B; " \
+		"for reg_addr in ${nrt_led_addrs2}; do i2c mw ${led_ctrl} ${reg_addr} 0x0A; done; " \
+		"for reg_addr in ${nrt_led_addrs1}; do i2c mw ${led_ctrl} ${reg_addr} 0xFF; done;\0" \
+	"start_led=run nrt_led_init; run nrt_led_start;\0" \
 	"loadm4bin=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bootdir}/${m4_bin}; " \
 		"cp.b ${loadaddr} ${m4_addr} ${filesize}\0" \
 	"runm4bin=" \
@@ -171,7 +179,7 @@
 #define CONFIG_BOOTCOMMAND \
 	"run ramsize_check; " \
         "setenv resin_kernel_load_addr ${loadaddr};" \
-        "run resin_set_kernel_root; run set_os_cmdline; gpio set 137; " \
+        "run resin_set_kernel_root; run start_led; run set_os_cmdline; gpio set 137; " \
         "setenv mmcdev ${resin_dev_index};" \
         "setenv mmcbootpart ${resin_boot_part}; run setconsole;" \
 	"mmc dev ${mmcdev}; "\
-- 
2.17.1

